### PR TITLE
Odim 5623: Support for Update BMC account under Managers

### PR DESCRIPTION
**Pull request by lpooja**

### CODE REVIEW

**Review by jeevan-kamkar — APPROVED** [review]

**Review by shivacharanms — APPROVED** [review]

File: svc-api/handle/managers.go [hunk 1]
```diff
@@ -334,6 +335,64 @@ func (mgr *ManagersRPCs) CreateRemoteAccountService(ctx iris.Context) {
...
+		log.Error(errorMessage)
+		response := common.GeneralError(http.StatusBadRequest, response.MalformedJSON, errorMessage, nil, nil)
+		common.SetResponseHeader(ctx, response.Header)
+		ctx.StatusCode(http.StatusBadRequest)
+		ctx.JSON(&response.Body)
+		return
+	}
+	request, err := json.Marshal(reqIn)
+	if err != nil {
+		errorMessage := "while trying to create JSON request body in create remote account: " + err.Error()
```
Comment (irappa-pattar) [inline svc-api/handle/managers.go]: update remote account

File: svc-managers/managers/managers.go [hunk 2]
```diff
@@ -679,6 +680,68 @@ func replaceBMCAccResp(data, managerID string) string {
...
+	} else if invalidProperties != "" {
+		errorMessage := "one or more properties given in the request body are not valid, ensure properties are listed in uppercamelcase "
+		log.Error(errorMessage)
+		response := common.GeneralError(http.StatusBadRequest, response.PropertyUnknown, errorMessage, []interface{}{invalidProperties}, nil)
+		return response
+	}
+
+	requestBody, err = json.Marshal(bmcAccReq)
+	if err != nil {
+		log.Error("while marshalling the create BMC account request: " + err.Error())
```
Comment (irappa-pattar) [inline svc-managers/managers/managers.go]: update account

File: svc-managers/mgrmodel/manager.go [hunk 3]
```diff
@@ -148,6 +148,13 @@ type CreateBMCAccount struct {
 	RoleID   string `json:"RoleId" validate:"required"`
 }
 
+// UpdateBMCAccount struct is to store the update BMC account request payload
+
+type UpdateBMCAccount struct {
+	Password string `json:"Password"`
```
Comment (irappa-pattar) [inline svc-managers/mgrmodel/manager.go]: add omitempty for both members

**Review by irappa-pattar — CHANGES_REQUESTED** [review]

File: svc-api/handle/managers.go (hunk 1 above)
Comment (lpooja) [reply svc-api/handle/managers.go]: addressed

File: svc-managers/managers/managers.go (hunk 2 above)
Comment (lpooja) [reply svc-managers/managers/managers.go]: addressed

**Review by irappa-pattar — APPROVED** [review]

**Review by amar-shalgar — APPROVED** [review]
LGTM

**Pull request merged**